### DISCUSSION TITLE
EMQX Edge bridge mode - [error] [Bridge] Can't be found from the inflight'

### DISCUSSION THREAD

**Discussion by gercorri** [emqx/emqx#4410] [Q&A] [answered]
Hi, has anyone been able to successfully run EMQX Edge in bridge mode?

We have it runinng on multiple device where Telegraf is configured to write device metrics (CPU, Mem) every 5 mins gets and a custome application is writing one message to a topic every 60sec. 

No matter what we do with the config when the EMQX brdige plugin is enable to forward the topics to the cloud EMQX broker it goes into a loop with error messages in the log saying '[error] [Bridge] Can't be found from the inflight'.  This seems to happen the emqx is in bridge mode on a device and there are other containers on the device subscribing and publishing to topics on it.

I can see two open github issues which seem to be the same and relate to this error - [error] [Bridge] Can't be found from the inflight':
https://github.com/emqx/emqx/issues/3629#issue-666217993 - milestone is set to 4.3-rc.1
https://github.com/emqx/emqx-bridge-mqtt/issues/81 - milestone set to 5.0-beta.1

I would like to know if anyone has a workaround for this to know if a timescale as to when fixes may be in place?
We have a SaaS product relying EMQX Edge which is due to into beta phase for 1st customers in a few weeks and this issue is a major blocker.

**Comment by zmstone** [comment]
@gercorri which version are you running?

@HJianBo 
seems to be a bug here:
https://github.com/emqx/emqx/blob/a4b30ea77cda182b2e158594af5c5a4f1236a15e/apps/emqx_bridge_mqtt/src/emqx_bridge_worker.erl

I am still not familiar with the code, how is ack references managed when replayq is in use?

**Comment by zmstone** [comment]
@gercorri could you send the configs ?

**Reply by gercorri** [reply]
Hi @zmstone we are trying some more changes to the config today. If thiose don't work then i'll upload what we've done so far

**Comment by gercorri** [comment]
Hi @zmstone and @HJianBo see attached config for the bridge. I have added a mountpoint (previously we had this commented out) and added in the relayq lines but it hasn't made a any difference. The main emqx.conf is default, we have made no changes there.

The issue appears to be totally random, a device could be running fine for two hours and then all of sudden develop the inflight issue.

In our current test setup we have 4 devices all Intel NUC running balenaOS. Each has telelgraf running which is just collecting device metrics and writing them to the bridge ever 5mins. We then have a custom app doing some video analytics which writes a message to one topic every 5secs.

I have change to the mqtt protocol setting to each of three version but it doesnt make any difference.

Each device is writing to the same mountpoint on the EQMX broker in AWS (we idenitify messages by unique tags per device).......would this be an issue?

Please review the config below and advice if there any settings you think we could modify that may help?

Are there any settings in the emqx.conf that could have an impact?

```
`##====================================================================
## Configuration for EMQ X MQTT Broker Bridge
##====================================================================

##--------------------------------------------------------------------
## Bridges to aws
##--------------------------------------------------------------------

## Bridge address: se node name for rpc bridging, use host:port for mqtt connection
##
## Value: String
## Example: emqx@127.0.0.1,  127.0.0.1:1883
## bridge.mqtt.aws.address = 127.0.0.1:1883
bridge.mqtt.aws.address = emqx.dev.xxxxxxx.xx:8883

## Protocol version of the bridge.
##
## Value: Enum
## - mqttv5
## - mqttv4
## - mqttv3
bridge.mqtt.aws.proto_ver = mqttv3

## Start type of the bridge.
##
## Value: enum
## manual
## auto
bridge.mqtt.aws.start_type = auto

## Whether to enable bridge mode for mqtt bridge
##
## This option is prepared for the mqtt broker which does not
## support bridge_mode such as the mqtt-plugin of the rabbitmq
##
## Value: boolean
# bridge.mqtt.aws.bridge_mode = true

## The ClientId of a remote bridge.
##
## Value: String
# bridge.mqtt.aws.clientid = test_device

## The Clean start flag of a remote bridge.
##
## Value: boolean
## Default: true
##
## NOTE: Some IoT platforms require clean_start
##       must be set to 'true'
bridge.mqtt.aws.clean_start = true

## The username for a remote bridge.
##
## Value: String
bridge.mqtt.aws.username = xxxxxxxxxxxxxx

## The password for a remote bridge.
##
## Value: String
bridge.mqtt.aws.password =xxxxxxxxxxxx

## Topics that need to be forward to aws IoTHUB
##
## Value: String
## Example: topic1/#,topic2/#                                                 
bridge.mqtt.aws.forwards = test_data/#,cpu,disk,diskio,mem,net,processes,swap,system

## Mountpoint of the bridge
bridge.mqtt.aws.forward_mountpoint = edge_devices/

## Need to subscribe to aws topics
##
## Value: String
## bridge.mqtt.aws.subscription.1.topic = cmd/topic1

## Need to subscribe to aws topics QoS.
##
## Value: Number
## bridge.mqtt.aws.subscription.1.qos = 1

## A mountpoint that receives messages from aws IoTHUB
##
## Value: String
## bridge.mqtt.aws.receive_mountpoint = receive/aws/


## Bribge to remote server via SSL.
##
## Value: on | off
bridge.mqtt.aws.ssl = on

## PEM-encoded CA certificates of the bridge.
##
## Value: File
bridge.mqtt.aws.cacertfile = /opt/emqx/etc/bridge_certs/cacert.pem

## Client SSL Certfile of the bridge.
##
## Value: File
bridge.mqtt.aws.certfile = /opt/emqx/etc/bridge_certs/client-cert.pem

## Client SSL Keyfile of the bridge.
##
## Value: File
bridge.mqtt.aws.keyfile = /opt/emqx/etc/bridge_certs/client-key.pem

## SSL Ciphers used by the bridge.
##
## Value: String
# bridge.mqtt.aws.ciphers = ECDHE-ECDSA-AES256-GCM-SHA384,ECDHE-RSA-AES256-GCM-SHA384

## Ciphers for TLS PSK.
## Note that 'bridge.${BridgeName}.ciphers' and 'bridge.${BridgeName}.psk_ciphers' cannot
## be configured at the same time.
## See 'https://tools.ietf.org/html/rfc4279#section-2'.
# bridge.mqtt.aws.psk_ciphers = PSK-AES128-CBC-SHA,PSK-AES256-CBC-SHA,PSK-3DES-EDE-CBC-SHA,PSK-RC4-SHA

## Ping interval of a down bridge.
##
## Value: Duration
## Default: 10 seconds
bridge.mqtt.aws.keepalive = 60s

## TLS versions used by the bridge.
##
## Value: String
bridge.mqtt.aws.tls_versions = tlsv1.2,tlsv1.1,tlsv1

## Bridge reconnect time.
##
## Value: Duration
## Default: 30 seconds
bridge.mqtt.aws.reconnect_interval = 30s

## Retry interval for bridge QoS1 message delivering.
##
## Value: Duration
bridge.mqtt.aws.retry_interval = 20s

## Publish messages in batches, only RPC Bridge supports
##
## Value: Integer
## default: 32
bridge.mqtt.aws.batch_size = 32

## Inflight size.
##
## Value: Integer
bridge.mqtt.aws.max_inflight_size = 32

## Base directory for replayq to store messages on disk
## If this config entry is missing or set to undefined,
## replayq works in a mem-only manner.
##
## Value: String
bridge.mqtt.aws.queue.replayq_dir = /opt/emqx/data/emqx_aws_bridge/

## Replayq segment size
##
## Value: Bytesize
bridge.mqtt.aws.queue.replayq_seg_bytes = 10MB

## Replayq max total size
##
## Value: Bytesize
bridge.mqtt.aws.queue.max_total_size = 1GB`
```

**Reply by gercorri** [reply]
We are running version 4.2.8

**Comment by zmstone** [comment]
@gercorri Thank you for the information. And sorry for the late reply.
We have a internal Jira ticket being followed up by @tigercl 
@tigercl Please keep this discussion thread updated too.

**Reply by zmstone** [reply]
fixed in this PR: https://github.com/emqx/emqx/pull/4513

**Reply by zmstone** [reply]
another fix: https://github.com/emqx/emqx/pull/4526

**Comment by zmstone** [comment, marked as answer]
To sum up: #4513 fixed the excessive logging issue, #4526 fixes the cause of the logging issue.

this fix is also ported back to 4.2.10
PR: emqx/emqx-bridge-mqtt#132
